userspace/libsinsp/parsers.cpp

```diff
@@ -159,6 +159,7 @@ void sinsp_parser::process_event(sinsp_evt *evt)
 		{
 			sinsp_evt_param *parinfo = evt->get_param(0);
 			uint16_t evid = *(uint16_t *)parinfo->m_val;
+			evt->betoh(&(evid));
```

Comment:
How did we end up needing to handle this in sinsp as well? :(

Also, why only a few events, like clone/execve? Doesn't this need to be applied to every integer parameter for every event?

Can we do this fixup in scap instead? Based on the event schema and the param lengths, we should be able to call the right conversion function for each param (none for char buffers, for example). The conversion would only need to happen while reading (non-native) captures, so the perf hit there shouldn't hurt too much.

OTOH, I believe this would be the first time we seriously need the event schema in libscap for nontrivial things though (it generally used to just shuffle byte buffers around). It also raises questions for plugins (how do we handle endianness when we don't have the schema?).